api/client/network.go

```diff
@@ -1,14 +1,304 @@
-// +build experimental
-
 package client
 
 import (
-	nwclient "github.com/docker/libnetwork/client"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"text/tabwriter"
+
+	"github.com/docker/docker/api/types"
+	Cli "github.com/docker/docker/cli"
+	flag "github.com/docker/docker/pkg/mflag"
+	"github.com/docker/docker/pkg/parsers/filters"
+	"github.com/docker/docker/pkg/stringid"
+)
+
+type command struct {
+	name        string
+	description string
+}
+
+var (
+	networkCommands = []command{
+		{"create", "Create a network"},
+		{"connect", "Connect container to a network"},
+		{"disconnect", "Disconnect container from a network"},
+		{"inspect", "Display detailed network information"},
+		{"ls", "List all networks"},
+		{"rm", "Remove a network"},
+	}
 )
 
-// CmdNetwork is used to create, display and configure network endpoints.
+// CmdNetwork is the parent subcommand for all network commands
+//
+// Usage: docker network <COMMAND> [OPTIONS]
 func (cli *DockerCli) CmdNetwork(args ...string) error {
-	nCli := nwclient.NewNetworkCli(cli.out, cli.err, nwclient.CallFunc(cli.callWrapper))
-	args = append([]string{"network"}, args...)
-	return nCli.Cmd("docker", args...)
+	cmd := Cli.Subcmd("network", []string{"COMMAND [OPTIONS]"}, networkUsage(), false)
+	cmd.Require(flag.Min, 1)
+	err := cmd.ParseFlags(args, true)
+	if err == nil {
```

Comment:
since `network` is a top level command with mandatory sub-commands, `ParseFlags` will handle the help usage if the user executes `docker network` without any argument.
But when the user executes an invalid argument such as `docker network foo`, `ParseFlags` cannot catch an error and hence it will return err as nil. But, since there is no corresponding `CmdNetworkFoo` handler, we must handle this condition under the `err == nil` condition.

But, it is infact, okay and generic to infact remove such a check all together and it is going to work fine.